src/bacnet/rpm.c

```diff
@@ -739,24 +767,42 @@ void rpm_ack_object_property_process(
                     apdu += len;
                 } else {
                     /* malformed */
+                    rp_data->error_class = ERROR_CLASS_SERVICES;
+                    rp_data->error_code = ERROR_CODE_INVALID_TAG;
+                    if (callback) {
+                        callback(device_id, rp_data);
+                    }
                     return;
                 }
                 if (bacnet_is_closing_tag_number(apdu, apdu_len, 5, &len)) {
                     apdu_len -= len;
                     apdu += len;
                 } else {
                     /* malformed */
+                    rp_data->error_class = ERROR_CLASS_SERVICES;
+                    rp_data->error_code = ERROR_CODE_INVALID_TAG;
+                    if (callback) {
+                        callback(device_id, rp_data);
+                    }
                     return;
                 }
                 if (callback) {
                     callback(device_id, rp_data);
                 }
             }
         }
-        len = rpm_decode_object_end(apdu, apdu_len);
-        if (len) {
+        /*  list-of-results [1] SEQUENCE OF SEQUENCE */
+        if (bacnet_is_closing_tag_number(apdu, apdu_len, 1, &len)) {
```

Comment:
As far as I understand, you can reach here only if `apdu_len` = 0, as there is no other way to exit from the loop above (`while(apdu_len)`) rather than having `apdu_len` = 0.